pkg/utils/functional/functional.go

```diff
@@ -89,3 +89,12 @@ func HasAnyPrefix(s string, prefixes ...string) bool {
 	}
 	return false
 }
+
+// InvertStringMap swaps keys to values and values to keys
```

Comment:
I'd add a warning that this should only be used on maps where all the values are distinct